### PR TITLE
Finishing experimental Gulp implementation

### PR DESCRIPTION
**Pull request by hermansje**
I solved the merge conflicts of pending pull request https://github.com/yeoman/generator-angular/pull/672 and meanwhile updated some gulp dependencies (clean -> rimraf, ngmin -> ngAnnotate) and adjusted the code to the changed file structure.

### CODE REVIEW

**Comment by esetnik** [comment]
I'd like to see this get merged in as well. The most popular angular generator should probably support the two most popular build systems. I don't feel it necessary to break this into two projects as we can cleanly encapsulate the build logic without creating a maintainability headache.

**Comment by arthurvr** [comment]
Honestly, at this point I'd even say it makes sense to drop Grunt in favor of Gulp. I don't see any reason not to. Also makes it way easier to maintain. I wonder what @eddiemonge thinks.

**Comment by stevemao** [comment]
:+1:

**Comment by eddiemonge** [comment]
I still like Grunt. I would prefer to see Grunt as the default and Gulp as an advanced option (not a prompt). I'd also want to make sure all the tools that are used for angular (some that aren't in here yet that I use daily) are supported in both first.

**Comment by eddiemonge** [comment]
can you rebase and resolve the conflicts please?

**Comment by hermansje** [comment]
I solved the conflicts and rebased the pull request.

File: app/index.js [hunk 1]
```diff
@@ -150,15 +150,43 @@ Generator.prototype.welcome = function welcome() {
...
+    message: 'Would you like to use Gulp (experimental) instead of Grunt?',
+    default: false
+  }], function (props) {
+    this.gulp = props.gulp;
+
+    cb();
+  }.bind(this));
+};
+
+Generator.prototype.askForStyles = function askForStyles() {
```
Comment (eddiemonge) [inline app/index.js]: this shouldnt be a part of this PR

**Pull request merged**